internal/util/profiles.go

```diff
@@ -0,0 +1,159 @@
+
+	var hashInput = ""
+	switch v := dataBlock.(type) {
+	case []string:
+		hashInput = strings.Join(v, "\n")
+	case string:
+		hashInput = v
+		// for all others, use string of encoded json to create has value
+	default:
+		Debug.Printf("unknown type for dataBlock, : %T\n", dataBlock)
+		json, err := json.Marshal(dataBlock)
+		// return nil Resultm if marshal failed
+		if err != nil {
+			return Result{}, err
+		}
+		hashInput = string(json)
+	}
+	// Get sha256 dataBlock
+
+	cacheId := CalcSha256(hashInput)
+
+	var profile Profile
+	profile.ProfileID = cacheId
+
+	savedCacheId := GetCacheValue(cacheFile)
+	if savedCacheId != cacheId {
+		profile.ProfileData = dataBlock
+		if updateCache {
+			Debug.Print("updating: ", cacheFile)
+			err := PutCacheValue(cacheFile, cacheId)
+			// return nill result if Put fails.
+			if err != nil {
+				return Result{}, err
+			}
+		}
+	}
+	return Result{tag: profile}, nil
+}
+
+func GetCacheValue(chkSumFile string) string {
+	// Only file nae passed in, need full file path
+	chkSumFilePath := profileFilePath + chkSumFile
+
+	// Before system registration chkSumFilePath won't exist
+	// in that case just return emtpy string
+	if !FileExists(chkSumFilePath) {
+		return ""
+	}
+	// Read value in from checksum file
+	content, err := os.ReadFile(chkSumFilePath)
+	// If read fails return empty string, should not happed
+	if err != nil {
+		return ""
+	}
+	return string(content)
+}
+
+func PutCacheValue(chkSumFile string, value string) error {
+	path := filepath.Dir(profileFilePath)
+	perms := os.FileMode(0755)
+	err := os.MkdirAll(path, perms)
+	if err != nil {
+		return err
+	}
+
+	// Only file name passed in, need full file path
+	chkSumFilePath := profileFilePath + chkSumFile
+	perms = os.FileMode(0644)
+	err = os.WriteFile(chkSumFilePath, []byte(value), perms)
+	return err
+}
+
+func DeleteProfileCache() {
+	// Remove profileFilePath and all files, ignore error
+	_ = os.RemoveAll(profileFilePath)
+}
+
+func CalcSha256(input string) string {
+	mkSha256 := sha256.New()
+	mkSha256.Write([]byte(input))
+	sha256 := mkSha256.Sum(nil)
+	return hex.EncodeToString(sha256)
+}
+
+func SetProfileFilePath(newPath string) {
+	profileFilePath = newPath
+	if profileFilePath[len(profileFilePath)-1] != '/' {
+		profileFilePath = profileFilePath + "/"
+	}
+}
+
+func GetProfileFilePath() string {
+	return profileFilePath
+}
+
+// SortAndUnique takes a byte array as input. This is to
+// conform with the return of util.Execute
+// it scans though the input, removes duplicated and items and sorts
+// the result. The intended use is to process things like lsmod data or rpm -qa
+// data to an uniform view of the data.
+func SortAndUnique(inputStream []byte) ([]string, error) {
+	scanner := bufio.NewScanner(bytes.NewReader(inputStream))
+
+	// Iterate through each line get module name
+	itemList := make(map[string]bool)
+	// Skip header line
```

Comment:
This seems implementation case specific, i.e. skipping a header row, and might be better managed via parameters, e.g. supporting both head and tail skip counts.